hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -1618,6 +1619,94 @@ class TestMORDataSource extends HoodieSparkClientTestBase with SparkDatasetMixin
     assertEquals(metaClient.getTableConfig.getRecordMergeStrategyId, mergerStrategyName)
   }
 
+  @Test
+  def testPayloadClassUpdate(): Unit = {
```

Comment:
I will parameterize these tests later - just want to put up a PR for early feedback.